pyracf/data_set/data_set_admin.py

```diff
@@ -110,15 +110,18 @@ def add(
             self._build_xml_segments(data_set_request)
             return self._make_request(data_set_request)
         try:
-            self.extract(data_set, volume=volume, generic=generic)
+            profile = self.extract(
```

Comment:
LISTDSD and RLIST commands both have the capability to fall back to a more generic profile than the one you specified. RLIST does this by default, LISTDSD does not. From our perspective, this means that you can functionally get a value from a profile extract for a profile you were not looking for. I discovered this when trying to add a started task, and I was getting AddOperationError because the profile ** already existed so it was coming back from these profile extracts. This check validates the name is what is expected before throwing the error.

Reply:
I also did add a unit test for this in Resource Admin to verify that Add and alter now avoid improperly basing information on the wrong profile. Feel free to remove the logic and run the tests to see the failures.

Reply:
Thank you for the explanation. I understand the purpose of this change now.